doc/getting_started/index.rst

```diff
@@ -367,31 +370,6 @@ are properly set up for Zephyr development.
       compatible with your board, a good alternative to try is the
       :ref:`Hello World sample <hello_world>`.
 
-#. Set build environment variables:
```

Comment:
Byeeeeeee :+1:

Reply:
The best path of the PR 😉

Reply:
We source `zephyr-env.sh` since it includes `~/.zephyrrc` where many people define `ZEPHYR_TOOLCHAIN_VARIANT`, `ZEPHYR_SDK_INSTALL_DIR` environment variables. I might have missed something obvious but how the build system is going to find the location of Zephyr SDK? The "Environment Variables" page still suggests using `~/.zephyrrc` file as one of the options where those environment variables can be set.

Reply:
> We source zephyr-env.sh since it includes ~/.zephyrrc where many people define ZEPHYR_TOOLCHAIN_VARIANT, ZEPHYR_SDK_INSTALL_DIR environment variables.

and that is still possible, but it is no longer mandatory.

This PR still allows the use of ZEPHYR_BASE and sourcing of `zephyr-env.sh`, but it removes this as a requirement to all users, and making it optional. Thus the PR is fully compatible with your current way of working.

But this PR should make it easier for new comers, and especially Windows users to get started.
It makes it easier for those using IDEs, with applications based on different Zephyr installations.
It also removes the issues stated in top of the PR.

The `getting_started` guide does not refer to `~/.zephyrrc`, it only refers to setting toolchain env vars directly in shell.
Nothing has changed in this regard.
https://github.com/zephyrproject-rtos/zephyr/blob/master/doc/getting_started/index.rst#L324-L325
```
      #. Set environment variables to let the build system know where to
         find the toolchain programs:

         .. code-block:: bash
            export ZEPHYR_TOOLCHAIN_VARIANT=zephyr
            export ZEPHYR_SDK_INSTALL_DIR=~/zephyr-sdk-0.11.2
```

But I agree, I should update also update this page:
https://docs.zephyrproject.org/latest/guides/env_vars.html

One way, could be add the possibility of adding `~/.zephyrrc` to `~/.bashrc` (or whatever shell people are using).

As I see it, sourcing `~/.zephyrrc` from `~/.bashrc` is just as good, cause even if having multiple Zephyr installations using different `zephyr-env.sh` would still end up in sourcing the same `~/.zephyrrc` and thus use the same toolchain settings.

Reply:
> > We source zephyr-env.sh since it includes ~/.zephyrrc where many people define ZEPHYR_TOOLCHAIN_VARIANT, ZEPHYR_SDK_INSTALL_DIR environment variables.
> 
> and that is still possible, but it is no longer mandatory.
> 
> This PR still allows the use of ZEPHYR_BASE and sourcing of `zephyr-env.sh`, but it removes this as a requirement to all users, and making it optional. 

This PR deals only with `ZEPHYR_BASE` but the user still needs to take care of setting `ZEPHYR_TOOLCHAIN_VARIANT` and on Linux `ZEPHYR_SDK_INSTALL_DIR` so we can't simply remove the whole "Set build environment variables:" step. To put it differently, setting `ZEPHYR_BASE` is not the only job done by `source zephyr-env.sh`. Rather than removing we should rewrite this part a bit better, explaining shortly the options and probably sending user to the "Environment Variables" page for more in depth explanation. Without this step the build will not work.

> As I see it, sourcing ~/.zephyrrc from ~/.bashrc is just as good, cause even if having multiple Zephyr installations using different zephyr-env.sh you would still end up in sourcing the same ~/.zephyrrc and thus use the same toolchain settings.

That's one of the methods proposed on "Environment Variables" page, certainly a valid one.

Reply:
> so we can't simply remove the whole "Set build environment variables:" step. 

Today there are two steps containing environment variables:
https://docs.zephyrproject.org/latest/getting_started/index.html#install-software-development-toolchain
Step 5.3
```
Set environment variables to let the build system know where to find the toolchain programs:

export ZEPHYR_TOOLCHAIN_VARIANT=zephyr
export ZEPHYR_SDK_INSTALL_DIR=~/zephyr-sdk-0.11.2
```

and step 6.1
```
Set build environment variables:

cd ~/zephyrproject/zephyr
source zephyr-env.sh
```

So this PR removes the `Set build environment variables` but keeps the `Set environment variables` step, cause that step is still required, as you can see.
![image](https://user-images.githubusercontent.com/9802583/77112265-fe8b4e80-6a28-11ea-82f2-dfca0ca77599.png)



So step 5.3 is kept, and the getting started guide mentions nothing regarding the placement of `ZEPHYR_TOOLCHAIN_VARIANT` and `ZEPHYR_SDK_INSTALL_DIR`.
That is only mentioned on "Environment Variables" page, so nothing has been changed on that part.

Of course, if a user places `ZEPHYR_TOOLCHAIN_VARIANT` and `ZEPHYR_SDK_INSTALL_DIR` in `~/.zephyrrc`, then he must also source `zephyr-env.sh`, but that is also described in the  "Environment Variables" page:
https://docs.zephyrproject.org/latest/guides/env_vars.html#option-3-using-zephyrrc-files
https://docs.zephyrproject.org/latest/guides/env_vars.html#zephyr-environment-scripts

> Rather than removing we should rewrite this part a bit better

That could be an option, but this PR is already fairly big, so I would prefer to keep the getting started as untouched as possible, and only address the bare minimum that I suggest to change for getting started.
But I think it could be very valuable to see if the getting started page could be improved in another PR.

> explaining shortly the options

For a getting started, we should keep things a simple as possible, that is, try to just explain one way of doing things.
Otherwise a first time user can easily be lost in what is required, or which way is the best.

> and probably sending user to the "Environment Variables" page for more in depth explanation. 

That is a very good suggestion.
Will add that to the `Set environment variables to let the build system know where to find the toolchain programs` step.


> > As I see it, sourcing ~/.zephyrrc from ~/.bashrc is just as good, cause even if having multiple Zephyr installations using different zephyr-env.sh you would still end up in sourcing the same ~/.zephyrrc and thus use the same toolchain settings.

> That's one of the methods proposed on "Environment Variables" page, certainly a valid one.

Exactly, so all 3 methods are described on "Environment Variables" and all three methods are still valid.
I'm sorry, but i'm not sure I fully understand what you are requesting, except that a reference from the "getting started" to the "environment variables" page is useful.

Getting started describes what is required, and has never mentioned where to place  `ZEPHYR_TOOLCHAIN_VARIANT` and `ZEPHYR_SDK_INSTALL_DIR`.
If user places those vars in `~/.zephyrrc`, then sourcing `zephyr-env.sh` could be one way to ensure those vars are included, as it is described on "Environment Variables" page.

Reply:
Setting up of environment variables hasn't been very well explained before and this PR makes it even more unclear.

```
Set environment variables to let the build system know where to find the toolchain programs:
```
In no way explains the user that it's their responsibility to make sure the variables are set the next time they open the terminal. After reading the Getting Started Guide user work environment should be all set up and the user should have a basic idea of how to use it. This document makes no mentions of what choices are there to set up the environment variables nor where to find such information. Without the explanation or the link to "Environment Variables" page we cannot assume user will know what's written there. This is pretty much the first page user who's new to the project will read.

In any case, I don't want to block this PR. I really like it otherwise. We can fix the documentation in the follow up PR.

Reply:
@mnkp Now I understand. You actually suggest that I improve the existing documentation, while i'm making this major change anyway.

That's a very good suggestion, and I agree with you, that getting started should really ensure that user's understand the system.

Let me see what I can do, but it might be after merging this PR.
I actually think we might be able to use the CMake package system for the toolchain as well, so that only user's who has special needs has to set `ZEPHYR_TOOLCHAIN_VARIANT` and `ZEPHYR_SDK_INSTALL_DIR` themselves, for example when using a 3rd party toolchain.

Other thoughts ?

Reply:
> I actually think we might be able to use the CMake package system for the toolchain as well, so that only user's who has special needs has to set `ZEPHYR_TOOLCHAIN_VARIANT` and `ZEPHYR_SDK_INSTALL_DIR` themselves, for example when using a 3rd party toolchain.

That would be nice!